docs/dialect/ops_reference.rst

```diff
@@ -31,45 +32,1304 @@ The `CSR64` encoding is usually defined with the alias::
+have the same shape as the expected output.
+
+The semiring must be a string of the form "<ADD_NAME>_<MUL_NAME>", e.g. "plus_times".
+The options for "<ADD_NAME>" are "plus", "any", and "min". The options for
+"<MUL_NAME>" are "pair", "times", "plus", "first", and "second". 
+
+If the first input is a matrix, it must be CSR format. If the second input
+is a matrix, it must be CSC format.  Matrix times vector will return a vector.
+Vector times matrix will return a vector.  Matrix times matrix will return
+a CSR matrix.
+
+The mask (if provided) must be the same format as the returned object. There's an optional
+boolean ``mask_complement`` attribute (which has a default value of ``false``) that will make
+the op use the complement of the mask.
+	
+Example:
+^^^^^^^^
+
+.. code-block:: text
+
+    %answer = graphblas.matrix_multiply %argA, %argB { semiring = "plus_plus" } : (tensor<?x?xi64, #CSR64>, tensor<?x?xi64, #CSC64>) to tensor<?x?xi64, #CSR64>
+
+.. code-block:: text
+
+    %answer = graphblas.matrix_multiply %argA, %argB, %mask { semiring = "min_times" } : (tensor<?x?xf64, #CSR64>, tensor<?x?xf64, #CSC64>, tensor<?x?xf64, #CSR64>) to tensor<?x?xf64, #CSR64>
+
+.. code-block:: text
+
+    %answer = graphblas.matrix_multiply %mat, %vec, %mask { semiring = "any_first", mask_complement = true } : (tensor<?x?xf64, #CSR64>, tensor<?xf64, #CV64>, tensor<?x?xf64, #CSR64>) to tensor<?x?xf64, #CSR64>
+
+.. code-block:: text
+
+    %answer = graphblas.matrix_multiply %vec, %mat { semiring = "min_second", mask_complement = true } : (tensor<?xf64, #CV64>, tensor<?x?xf64, #CSR64>) to tensor<?x?xf64, #CSR64>
+
+.. code-block:: text
+
+    %answer = graphblas.matrix_multiply %vecA, %vecB, %mask { semiring = "any_pair" } : (tensor<?xf64, #CV64>, tensor<?xf64, #CV64>, tensor<?xf64, #CV64>) to tensor<?xf64, #CV64>
+
+Syntax:
+^^^^^^^
+
+.. code-block:: text
+
+    operation ::= `graphblas.matrix_multiply` $a `,` $b (`,` $mask^)? attr-dict `:` `(` type($a) `,` type($b)  (`,` type($mask)^)? `)` `to` type($output)
+
+Attributes:
+^^^^^^^^^^^
+
+.. list-table::
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Attribute
+     - MLIRType
+     - Description
+   * - ``semiring``
+     - ``::mlir::StringAttr``
+     - Semiring.
+   * - ``mask_complement``
+     - ``::mlir::BoolAttr``
+     - (Optional) Whether to use the mask or the complement of the mask. Is false by default.
+
+Operands:
+^^^^^^^^^
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Operand
+     - Description
+   * - ``a``
+     - Left-hand matrix multiply tensor operand (CSR matrix or sparse vector)
+   * - ``b``
+     - Right-hand matrix multiply tensor operand (CSC matrix or sparse vector)
+   * - ``mask``
+     -  Optional structural mask (CSR matrix or sparse vector)
+
+Results:
+^^^^^^^^
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Result
+     - Description
+   * - ``output``
+     - Output tensor (CSR matrix or sparse vector).
```

Comment:
output can be CSR matrix, sparse vector, or scalar

Reply:
https://github.com/metagraph-dev/mlir-graphblas/commit/7b048414850981638a076247c253685efeb6250c implements this suggestion.